scripts/build/termux_step_get_dependencies.sh

```diff
@@ -57,21 +82,38 @@ termux_step_get_dependencies() {
 					fi
 				)
 			fi
-
 			mkdir -p $TERMUX_BUILT_PACKAGES_DIRECTORY
 			echo "$DEP_VERSION" > "$TERMUX_BUILT_PACKAGES_DIRECTORY/$PKG"
 		done<<<$(./scripts/buildorder.py -i "$TERMUX_PKG_BUILDER_DIR" $TERMUX_PACKAGES_DIRECTORIES || echo "ERROR")
-	elif [ "$TERMUX_SKIP_DEPCHECK" = false ] && [ "$TERMUX_INSTALL_DEPS" = false ] && [ "$TERMUX_PKG_METAPACKAGE" = "false" ]; then
+	else
 		# Build dependencies
 		while read PKG PKG_DIR; do
 			if [ -z $PKG ]; then
 				continue
 			elif [ "$PKG" = "ERROR" ]; then
 				termux_error_exit "Obtaining buildorder failed"
 			fi
-			echo "Building dependency $PKG if necessary..."
 			# Built dependencies are put in the default TERMUX_OUTPUT_DIR instead of the specified one
-			TERMUX_BUILD_IGNORE_LOCK=true ./build-package.sh $(test "${TERMUX_FORCE_BUILD}" = "true" && echo "-f" || true) -s --format $TERMUX_PACKAGE_FORMAT "${PKG_DIR}"
+			if [ "$TERMUX_FORCE_BUILD_DEPENDENCIES" = "true" ]; then
+				[ ! "$TERMUX_QUIET_BUILD" = true ] && echo "Force building dependency $PKG..."
+				if [ "$TERMUX_ON_DEVICE_BUILD" = "true" ] && ! pacakge__is_package_on_device_build_supported "$PKG_DIR"; then
+					echo "Building $PKG on device is not supported. Consider passing -I flag to download it instead"
+					return 1
+				fi
+				read DEP_ARCH DEP_VERSION DEP_VERSION_PAC <<< $(termux_extract_dep_info $PKG "${PKG_DIR}")
+				termux_force_check_package_dependency
+			else
+				[ ! "$TERMUX_QUIET_BUILD" = true ] && echo "Building dependency $PKG if necessary..."
+			fi
+			TERMUX_BUILD_IGNORE_LOCK=true ./build-package.sh -s $(test "${TERMUX_FORCE_BUILD_DEPENDENCIES}" = "true" && echo "-F" || true) --format $TERMUX_PACKAGE_FORMAT "${PKG_DIR}"
+			pacakge__add_package_to_built_packages_list "$PKG"
 		done<<<$(./scripts/buildorder.py "$TERMUX_PKG_BUILDER_DIR" $TERMUX_PACKAGES_DIRECTORIES || echo "ERROR")
 	fi
 }
+
+termux_force_check_package_dependency() {
+	if [ "$TERMUX_BUILD_PACKAGE_CALL_DEPTH" -gt 0 ] && [[ " $TERMUX_BUILD_PACKAGE_CALL_BUILT_PACKAGES_LIST " == *" $PKG "* ]] && [ -e "$TERMUX_BUILT_PACKAGES_DIRECTORY/$PKG" ] && [ "$(cat "$TERMUX_BUILT_PACKAGES_DIRECTORY/$PKG")" = "$DEP_VERSION" ]; then
+		[ ! "$TERMUX_QUIET_BUILD" = true ] && echo "Skipping already built dependency $PKG@$DEP_VERSION"
+		continue
```

Comment:
You have changed logic again from what I provided and no idea what you are trying to do with a `continue` statement inside a separate function outside a loop. It would throw errors too, I guess you are not testing either.

I also don't have time to test currently, packages people should do it.

Reply:
> You have changed logic again from what I provided. and no idea what you are trying to do with a continue statement inside a separate function outside a loop.

In this way, I tried to reduce the code from unnecessary duplication of `build-package.sh` launches. Also, I try to put the minimum number of functions in files like `termux_step_get_dependencies.sh` (I try to follow the termux scripting *standard*).

> Tha'ts from my code, use return 0. It would throw errors too, I guess you are not testing either.

I tested the compiler on the device and it works and compiles the packages successfully. As for testing in docker, I did not do it.

> I also don't have time to test currently, packages people should do it.

Does the compiler have secrets from us? 🤔 /j

Reply:
>In this way, I tried to reduce the code from unnecessary duplication of build-package.sh launches. Also, I try to put the minimum number of functions in files like termux_step_get_dependencies.sh (I try to follow the termux scripting standard).

It would still result in `1` function being added and would be more readable since force building will be handled in a single function and won't require multiple ifs. And the `termux_force_*` function is a sub function of `termux_step_get_dependencies`, that won't be used elsewhere, so in my opinion, its perfectly fine to add sub functions in the same file, since that's where they belong.

>Tha'ts from my code

Sorry about that again. 

>I tested the compiler on the device and it works and compiles the packages successfully. As for testing in docker, I did not do it

It should ideally be tested if packages are getting force built if not on device, specially ones not supported on device. 

>Does the compiler have secrets from us? thinking /j

Duh, have you never had the situation where compiler kept failing for hours and then magically worked the next morning.

Reply:
There is still a `continue` statement in the function. You haven't fixed it, so this would be a bug too.